diff --git a/src/backends/onnx.js b/src/backends/onnx.js
index 26d771b..17ad55c 100644
--- a/src/backends/onnx.js
+++ b/src/backends/onnx.js
@@ -20,7 +20,7 @@ import { env, apis } from '../env.js';

 // NOTE: Import order matters here. We need to import `onnxruntime-node` before `onnxruntime-web`.
 // In either case, we select the default export if it exists, otherwise we use the named export.
-import * as ONNX_NODE from 'onnxruntime-node';
+//import * as ONNX_NODE from 'onnxruntime-node';

 // Use subpath-imports to ensure Node.js and browser interoperability.
 // See package.json and https://nodejs.org/api/packages.html#subpath-imports
diff --git a/webpack.config.js b/webpack.config.js
index 2182492..8838d72 100644
--- a/webpack.config.js
+++ b/webpack.config.js
@@ -19,7 +19,8 @@ function buildConfig({
   suffix = ".js",
   type = "module", // 'module' | 'commonjs'
   ignoreModules = [],
-  externalModules = [],
+  externalModules = {},
+  entry = null,
 } = {}) {
   const outputModule = type === "module";

@@ -31,27 +32,26 @@ function buildConfig({

   /** @type {import('webpack').Configuration} */
   const config = {
-    mode: 'development',
-    devtool: 'source-map',
-    entry: {
-      [`transformers${name}`]: './src/transformers.js',
-      [`transformers${name}.min`]: './src/transformers.js',
-    },
+    mode: "development",
+    devtool: "source-map",
+    entry: entry,
     output: {
       filename: `[name]${suffix}`,
-      path: path.join(__dirname, 'dist'),
+      path: path.join(__dirname, "dist"),
       library: {
         type,
       },
-      assetModuleFilename: '[name][ext]',
-      chunkFormat: 'module',
+      assetModuleFilename: "[name][ext]",
+      chunkFormat: "module",
     },
     optimization: {
       minimize: true,
-      minimizer: [new TerserPlugin({
-        test: new RegExp(`\\.min\\${suffix}$`),
-        extractComments: false,
-      })],
+      minimizer: [
+        new TerserPlugin({
+          test: new RegExp(`\\.min\\${suffix}$`),
+          extractComments: false,
+        }),
+      ],
     },
     experiments: {
       outputModule,
@@ -73,12 +73,12 @@ function buildConfig({
     config.module = {
       parser: {
         javascript: {
-          importMeta: false
-        }
-      }
-    }
+          importMeta: false,
+        },
+      },
+    };
   } else {
-    config.externalsType = 'commonjs';
+    config.externalsType = "commonjs";
   }

   return config;
@@ -91,26 +91,29 @@ const NODE_IGNORE_MODULES = ["onnxruntime-web", "onnxruntime-web/webgpu"];
 // Do not bundle the following modules with webpack (mark as external)
 // NOTE: This is necessary for both type="module" and type="commonjs",
 // and will be ignored when building for web (only used for node/deno)
-const NODE_EXTERNAL_MODULES = ["onnxruntime-node", "sharp", "fs", "path", "url"];
-
+const NODE_EXTERNAL_MODULES = [
+  "onnxruntime-node",
+  "sharp",
+  "fs",
+  "path",
+  "url",
+];

 export default [
-  // Web-only build
   buildConfig({
     type: "module",
+    externalModules: {
+      "#onnxruntime-webgpu": "chrome://global/content/ml/ort.webgpu-dev.mjs",
+    },
+    entry: { ["transformers"]: "./src/transformers.js" },
+    ignoreModules: NODE_EXTERNAL_MODULES,
   }),
-
-  // Node-compatible builds
   buildConfig({
-    suffix: ".mjs",
     type: "module",
-    ignoreModules: NODE_IGNORE_MODULES,
-    externalModules: NODE_EXTERNAL_MODULES,
-  }),
-  buildConfig({
-    suffix: ".cjs",
-    type: "commonjs",
-    ignoreModules: NODE_IGNORE_MODULES,
-    externalModules: NODE_EXTERNAL_MODULES,
+    externalModules: {
+      "#onnxruntime-webgpu": "chrome://global/content/ml/ort.webgpu.mjs",
+    },
+    entry: { ["transformers.min"]: "./src/transformers.js" },
+    ignoreModules: NODE_EXTERNAL_MODULES,
   }),
 ];
